vortex-array/src/array/struct_/mod.rs

```diff
@@ -150,11 +150,24 @@ impl VariantsVTable<StructArray> for StructEncoding {
 
 impl StructArrayTrait for StructArray {
     fn maybe_null_field_by_idx(&self, idx: usize) -> Option<ArrayData> {
-        self.dtypes().get(idx).map(|dtype| {
-            self.as_ref()
-                .child(idx, dtype, self.len())
-                .unwrap_or_else(|e| vortex_panic!(e, "StructArray: field {} not found", idx))
-        })
+        Some(
+            self.field_info(&Field::Index(idx))
```

Comment:
I think this function is really really gross. However, I can't fix it without completely redoing how struct scalars are implemented. I took a swing at that but it got too massive so I left that on another branch and kept this as-is

Reply:
Basically, `maybe_null_field` (which IMO is poorly named) can return an Option<ArrayData>, but currently there are NO cases where it ever will return `None`, it will either return Some() or it will panic.

It doesn't even return `None` if you try and index to an invalid field, which is sort of why I thought this method was Option-typed to begin with. It will in fact just panic.

This flows all the way down to StructArray, StructDType, DType, StructScalar, Patches...you have to touch a lot of code to make this suck less.

Reply:
it should really return None for cases where the field is missing. This is really unfortunate